### ISSUE TITLE
Jetbrains IDEs Remote Development on WSL

### ISSUE THREAD

**Issue by Cyberavater** [nix-community/NixOS-WSL#321]
## Bug description

It seems any of the JetBrains IDEs cannot connect to NixOS on WSL
## To Reproduce
When I try to select NixOS from the WSL menu in IntelliJ nothing happens.
**Expected behavior**

But if I select Ubuntu, it downloads the backend and runs without any problems.
## Logs
```
Include relevant console logs
```

**Comment by DaRacci** [comment]
Don't bother with remote dev, its really buggy and it doesn't work due to missing items in /bin and other hardcoded locations, install rust rover on wsl directly and run it using wslg

**Comment by Cyberavater** [comment]
> Don't bother with remote dev, its really buggy and it doesn't work due to missing items in /bin and other hardcoded locations, install intellij on wsl directly and run it using wslg

Yes, I was thinking about doing that. Tell me your experience , was it smooth?

**Comment by DaRacci** [comment]
Yeah i actually only got it working like yesterday but its smooth as, besides the random freeze up every few hours, but i think thats just rust rover being in early access atm.
Just had to add /usr/lib/wsl/lib to my session variables so that gpu acceleration worked instead of software rendering.

**Comment by DaRacci** [comment]
Another thing is im using home-manager i also had to add my own script to add applications to the windows start menu like:
```
# Fixes Home-Manager applications not appearing in Start Menu
  system.activationScripts.copy-user-launchers = stringAfter [ ] ''
    for x in applications icons; do
      echo "setting up /usr/share/''${x}..."
      targets=()
      if [[ -d "/home/${config.wsl.defaultUser}/.nix-profile/share/$x" ]]; then
        targets+=("/home/${config.wsl.defaultUser}/.nix-profile/share/$x/.")
      fi

      if (( ''${#targets[@]} != 0 )); then
        mkdir -p "/usr/share/$x"
        ${pkgs.rsync}/bin/rsync -ar --delete-after "''${targets[@]}" "/usr/share/$x"
      else
        rm -rf "/usr/share/$x"
      fi
    done
  '';

**Comment by Cyberavater** [comment]
Thanks for your advice man! appreciate ya!

Another thing I'd like to ask is since you're using `home-manager` are you on `Ubuntu` or `NixOS` itself?

**Comment by DaRacci** [comment]
Im on pure nixos you can check my configurations here if you want a reference, mainly look at the [wsl.nix file](https://github.com/DaRacci/nix-config/blob/master/hosts/common/optional/wsl.nix) and then the [home-manager profile](https://github.com/DaRacci/nix-config/blob/master/home/racci/winix.nix)
